src/XProtocol/XProtocol.hh

```diff
@@ -1391,6 +1391,10 @@ static int mapError(int rc)
            case ETIMEDOUT:     return kXR_ReqTimedOut;
            case EBADF:         return kXR_FileNotOpen;
            case ECANCELED:     return kXR_Cancelled;
+           // In the case one tries to delete a non-empty directory
+           // we have decided that until the next major release
+           // the kXR_ItExists flag will be returned
+           case ENOTEMPTY:     return kXR_ItExists;
```

Comment:
What should happen here is that this case needs to be put ahead if the kXR_ItExists case as a falthrough.  That is
using [[fallthrough]]; to indicate that the fallthrough is correct.

Reply:
I do not understand :thinking: Is it a fallthrough case? I just return kXR_ItExists when the errno code is ENOTEMPTY. Each switch case has a return value, so there's no need for a `[[fallthrough]];` attribute, isn't it?